Bio/SeqIO/PdbIO.py

```diff
@@ -216,6 +216,31 @@ def iterate(self, handle):
                         "db_id_code": db_id_code,
                     }
                 )
+            elif rec_name == "DBREF1":
+                # ID code of this entry (PDB ID)
+                pdb_id = line[7:11]
+                # Chain identifier.
+                chn_id = line[12]
+                # Sequence database name.
+                database = line[26:32].strip()
+                # Sequence database identification code.
+                db_id_code = line[47:67].strip()
+                metadata[chn_id].append(
+                    {
+                        "pdb_id": pdb_id,
+                        "database": database,
+                        "db_id_code": db_id_code,
+                    }
+                )
+            elif rec_name == "DBREF2":
+                # ID code of this entry (PDB ID)
+                pdb_id = line[7:11]
+                # Chain identifier.
+                chn_id = line[12]
+                # Sequence database accession code.
+                db_acc = line[18:40].strip()
+                if chn_id in metadata:
+                    metadata[chn_id][0]["db_acc"] = db_acc
```

Comment:
There should be only one pair of ``DBREF1`` and ``DBREF2`` lines, right?

Currently you setup a partial annotation entry with the ``DBREF1`` line, and then add the accession when parsing the ``DBREF2`` line.

How about something like this:

```python
            elif rec_name == "DBREF1":
                # ID code of this entry (PDB ID)
                pdb_id = line[7:11]
                # Chain identifier.
                chn_id = line[12]
                # Sequence database name.
                database = line[26:32].strip()
                # Sequence database identification code.
                db_id_code = line[47:67].strip()
            elif rec_name == "DBREF2":
                # Check these are consistent:
                if pdb_id != line[7:11] or chn_id != line[12]:
                    raise ValueError("DBREF2 identifiers do not match")
                # This line has the sequence database accession code.
                metadata[chn_id].append(
                    {
                        "pdb_id": pdb_id,
                        "database": database,
                        "db_id_code": db_id_code,
                        "db_acc": line[18:40].strip(),
                    }
                )
```

Reply:
> There should be only one pair of `DBREF1` and `DBREF2` lines, right?

I believe `DBREF1` and `DBREF2` should come in pairs, right. There may be more pairs, but single chain should have at most one pair.

> Currently you setup a partial annotation entry with the `DBREF1` line, and then add the accession when parsing the `DBREF2` line.
> 
> How about something like this:

[skip]

I agree with your suggestion. Will amend the PR soon.